src/v2i-hub/SpatPlugin/src/PedestrianDetectionForSPAT.cpp

```diff
@@ -0,0 +1,50 @@
+#include "PedestrianDetectionForSPAT.h"
+
+#include <vector>
+
+using namespace tmx::messages;
+
+void PedestrianDetectionForSPAT::updateEncodedSpat(SpatEncodedMessage & spatEncodedMsg,
+        std::shared_ptr<SpatMessage> _spatMessage,
+        const std::string & currentPedLanes)  const
+{
+    // Add pedestrian lanes with active detections and clear the rest
+    auto spat = _spatMessage->get_j2735_data();
+    // check for valid SPAT and also if ped lanes is filled in
+    if (spat && spat->intersections.list.array && spat->intersections.list.count > 0 && currentPedLanes.length()) {
+        // parse ped zone string into set of int values
+        std::vector<LaneConnectionID_t> zones;
+        {
+            std::vector<char> zoneList(currentPedLanes.length());
+            std::copy(currentPedLanes.begin(), currentPedLanes.end(), zoneList.begin());
+            char *restOfString = nullptr;
+            auto c = strtok_r(zoneList.data(), ",", &restOfString);
+
+            while (c != nullptr) {
+                zones.push_back(strtol(c, nullptr, 0));
+                c = strtok_r(nullptr, ",", &restOfString);
+            }
+        }
+
+        if (!zones.empty()) {
+            ManeuverAssistList *&mas = spat->intersections.list.array[0]->maneuverAssistList;
+            mas = (ManeuverAssistList *) calloc(1, sizeof(ManeuverAssistList));
```

Comment:
I see there were some code smells for using calloc. I know this was just moved over from signalController.cpp. Do you think we can address those now or should we do it separately?

Reply:
I think for the J2735 encoding/decoding library there is not a way to use smart pointers instead of calloc calls